src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,336 @@
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </remarks>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometerWithoutCorrection(bool waitForData, TimeSpan timeout)
+        {
+            Span<byte> rawData = stackalloc byte[6];
+            // Wait for a data to be present
+            if (waitForData)
+            {
+                DateTime dt = DateTime.Now.Add(timeout);
+                while (!HasDataToRead)
+                {
+                    if (DateTime.Now > dt)
+                        throw new TimeoutException($"{nameof(ReadMagnetometer)} timeout reading value");
+                }
+            }
+
+            ReadBytes(Register.HXL, rawData);
+            // In continuous mode, make sure to read the ST2 data to clear up
+            if ((_measurementMode == MeasurementMode.ContinuousMeasurement100Hz) ||
+                (_measurementMode == MeasurementMode.ContinuousMeasurement8Hz))
+            {
+                ReadByte(Register.ST2);
+            }
+
+            Vector3 magneto = new Vector3();
+            magneto.X = BinaryPrimitives.ReadInt16LittleEndian(rawData);
+            magneto.Y = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(2));
+            magneto.Z = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(4));
+
+            if (OutputBitMode == OutputBitMode.Output16bit)
+            {
+                // From the documentation range is from 32760 which does represent 4912 µT
+                // result of 4912.0f / 32760.0f
+                magneto *= 0.1499389499389499f;
+            }
+            else
+            {
+                // result of 4912.0f / 8192.0f
+                magneto *= 0.599609375f;
+            }
+            return magneto;
+
+        }
+
+        /// <summary>
+        /// Read the magnetometer with bias correction and can wait for new data to be present
+        /// </summary>
+        /// <remarks>
+        /// Vector axes are the following:
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </remarks>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometer(bool waitForData, TimeSpan timeout)
+        {
+            var magn = ReadMagnetometerWithoutCorrection(waitForData, timeout);
+            magn.X *= MagnetometerHardwareAdjustmeent.X;
```

Comment:
I believe either `magn *= MagnetometerHardwareAdjustmeent;`. If that doesn't work then:
`magn = Vector3.Multiply(magn, MagnetometerHardwareAdjustmeent);`

Reply:
Yes, it's working. I was expecting another result as Vector and Matric multiplication are not that straight forward.